src/library/dao/directorydao.cpp

```diff
@@ -167,17 +165,13 @@ QList<RelocatedTrack> DirectoryDAO::relocateDirectory(
     // match both instead of only files in the parent directory "a/b/".
     DEBUG_ASSERT(!oldDirectory.endsWith('/'));
     const QString oldDirectoryPrefix = oldDirectory + '/';
-    const QString startsWithOldDirectory = SqlLikeWildcardEscaper::apply(
-                                                   oldDirectoryPrefix, kSqlLikeMatchAll) +
-            kSqlLikeMatchAll;
-
     query.prepare(QStringLiteral(
             "SELECT library.id,track_locations.id,track_locations.location "
             "FROM library INNER JOIN track_locations ON "
             "track_locations.id=library.location WHERE "
-            "track_locations.location LIKE :startsWithOldDirectory ESCAPE :escape"));
-    query.bindValue(":startsWithOldDirectory", startsWithOldDirectory);
-    query.bindValue(":escape", kSqlLikeMatchAll);
+            "SUBSTR(track_locations.location,1,:oldDirectoryPrefixLength)=:oldDirectoryPrefix"));
```

Comment:
We usually don't explain the obvious in comments, but in this case it might be good to add a comment that `1` refers to the first character in the string (i.e. it's not 0-based like arrays/lists/whatever in C/C++ are).

Reply:
Another way to implement something like `STARTSWITH()` would be `INSTR(track_locations.location, :oldDirectoryPrefix) = 1`, but I guess that would be less performant.

Reply:
I won't replicate the same comment at 3 different locations. This can be looked up in the SQLite docs.

Using INSTR is less chatty, we should probably use that!

Reply:
The naming of the variables clearly reveals the purpose, so there is no need to explain the 1-based indexing. 

Thinking about the performance implications SUBSTR might in fact be faster than INSTR, because you have to know the outer context for optimizing the INSTR operation. I will keep using it in the aoide backend.